Dynatrace/dynatrace-operator
# EdgeConnect pod spec enhancements for Kubernetes Automation (ServiceAccountName, HostAlias)

## Description

Sets the following fields of EdgeConnect pod spec :
- `serviceAccount`, `serviceAccountName` to _CR.Spec.ServiceAccount_
- `hostAliases[0].IP` to _ClusterIP_ of the _kubernetes.default_ service
- `hostAliases[0].Hostnames[0]` to _CR-name.CR-namespace.UID-of-the-kube-system-namespace_

Both fields are always set  (also if K8S Automation is disabled).

Operator needs new clusterrole to get `kubernetes.default` service.

## How can this be tested?

- Unittests
- Deploy EdgeConnect in normal mode and provisioner mode and verify spec of the EdgeConnect pod
- Deploy EdgeConnect with different ServiceAccount
```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: test-normal-edgeconnect
  namespace: dynatrace
  labels:
    app.kubernetes.io/name: edgeconnect
    app.kubernetes.io/version: "0.0.0-snapshot"
    app.kubernetes.io/component: user-provisioned
```
